uaa/build.gradle

```diff
@@ -55,7 +55,7 @@ dependencies {
     providedCompile(libraries.tomcatEmbed)
 
     implementation(libraries.springMeteringFilter) {
-        exclude(module: "javax.mail")
+        exclude(group: "javax.mail")
```

Comment:
Why do you think changing it from module to group would fix the exclusion?
Shouldn't the right format for exclusion be: `exclude(group: "com.sun.mail", module: "javax.mail")`

Also, I don't see that UAA pulls in `com.sun.mail:jakarta.mail`at runtime.  Yes, the library is declared [dependencies.gradle#L50](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/dependencies.gradle#L50), but it's only included as testImplementation [uaa/build.gradle#L104](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/uaa/build.gradle#L104)

Another point, we exclude by module for other dependencie. For example, [uaa/build.gradle#L35-L39](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/uaa/build.gradle#L35-L39).  Do you think that would be causing issues as well?

Reply:
As I mention in the commit message, the library that causes the conflict is javax.mail:mailapi (javax.mail is the group, mailapi is the module). We could be explicit and exclude only the module from spring-metering-filter, but usually when you include javax.mail:mailapi, you need to have an implementation for the API, which I think is javax.mail:mail. So guess its best to exclude all the modules in the group, rather than just the mailapi module. And that actually was how it was in the beginning. See https://github.com/GESoftware-CF/uaa/pull/48/commits/aff34e28ac9fe927e1dc18267d7649d4735e2096. But that got tweaked incorrectly in https://github.com/GESoftware-CF/uaa/commit/11777bb7911a29c89971aa4ef9bc52e778983752# - which changed the exclusion from group to module, but javax.mail is the group, not the module.

The server module pulls it in as an `implementation`: https://github.com/GESoftware-CF/uaa/blob/release_75.18.0/server/build.gradle#L16 - I don't think there is anything else that supports sending of emails. The JARs before this exclusion fix included in the WAR were jakarta.mail and mailapi. After this exclusion fix, its only jakarta.mail and it works.

You can exclude all the modules in a group, or just the module. In this case, think its best to exclude the whole group rather than just the mailapi module.

Reply:
Thanks for the explanation. That makes sense now.

Reply:
I see what you mean by com.sun.mail:javax.mail group/module. That is very confusing. At one point there was also a javax.mail:javax.mail group/module :( I have updated the commit message to make things clearer.